src/DTO/ProductDTO.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DTO;
+
+class ProductDTO {
+
+    protected string $id;
+    protected string $productName;
+    protected string $description;
+
+    public function __construct(array $data) {
+        $this->setId($data['id']);
```

Comment:
you don't do any checks if array have these fields -> in this case it is all good, but you could just check if this data is set in next task